sway/config/output.c

```diff
@@ -164,23 +164,31 @@ static void set_mode(struct wlr_output *output, int width, int height,
 	}
 }
 
-void terminate_swaybg(pid_t pid) {
-	int ret = kill(pid, SIGTERM);
-	if (ret != 0) {
-		wlr_log(WLR_ERROR, "Unable to terminate swaybg [pid: %d]", pid);
-	} else {
-		int status;
-		waitpid(pid, &status, 0);
+static void terminate_swaybg(struct wl_display *display, pid_t pid) {
+	struct wl_client *client;
+	wl_client_for_each(client, wl_display_get_client_list(display)) {
+		pid_t p;
+		wl_client_get_credentials(client, &p, NULL, NULL);
+
+		if (p == pid) {
+			wl_client_destroy(client);
+			break;
+		}
+	}
+
+	if (kill(pid, SIGTERM) != 0) {
```

Comment:
I would also do the kill before `wl_client_destroy` (before the whole loop?).
If a well-behaved client notices the destroy (that closes the socket connection right?) and exits before we send the kill, it's possible that the pid gets reused meanwhile which would make us kill a random process, or in most cases we could log an error.